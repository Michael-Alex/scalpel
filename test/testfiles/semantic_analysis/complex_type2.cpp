int ((*const*(*(*i))[3]))[5];

