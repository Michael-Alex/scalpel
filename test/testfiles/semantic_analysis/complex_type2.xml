<namespace id="n0">
 <variable name="i">
  <type>
   <array size="0">
    <array size="0">
     <pointer>
      <pointer>
       <pointer>
        <cv_qualified_type const="true">
         <pointer>
          <fundamental_type type="int"/>
         </pointer>
        </cv_qualified_type>
       </pointer>
      </pointer>
     </pointer>
    </array>
   </array>
  </type>
 </variable>
</namespace>
