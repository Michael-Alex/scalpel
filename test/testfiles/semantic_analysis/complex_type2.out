{
  "id": 0,
  "variables":
  [
    {
      "id": 0,
      "name": "i",
      "type":
      {
        "pointer":
        {
          "type":
          {
            "pointer":
            {
              "type":
              {
                "array":
                {
                  "size": 3,
                  "type":
                  {
                    "pointer":
                    {
                      "type":
                      {
                        "cv-qualified type":
                        {
                          "const": true,
                          "type":
                          {
                            "pointer":
                            {
                              "type":
                              {
                                "array":
                                {
                                  "size": 5,
                                  "type":
                                  {
                                    "fundamental type": "int"
                                  }
                                }
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  ]
}
